### PR TITLE
[bitnami/etcd] fix snapshots directory ownership when security context is enabled

### LINKED ISSUE #5759: [bitnami/etcd]snapshots directory permission denied
```
securityContext:
  enabled: true
  fsGroup: 1001
  runAsUser: 1001
```

```
==> Bash debug is on
+ AUTH_OPTIONS='--user root:puIV%G!QRU1%vjII'
+ export ETCDCTL_ENDPOINTS=etcd-0.etcd-headless.etcd-system.svc.cluster.local:2379,etcd-1.etcd-headless.etcd-system.svc.cluster.local:2379,etcd-2.etcd-headless.etcd-system.svc.cluster.local:2379
+ ETCDCTL_ENDPOINTS=etcd-0.etcd-headless.etcd-system.svc.cluster.local:2379,etcd-1.etcd-headless.etcd-system.svc.cluster.local:2379,etcd-2.etcd-headless.etcd-system.svc.cluster.local:2379
+ mkdir -p /snapshots
+ read -r -a endpoints
++ tr ',;' ' '
+ for endp in "${endpoints[@]}"
+ echo 'Using endpoint etcd-0.etcd-headless.etcd-system.svc.cluster.local:2379'
Using endpoint etcd-0.etcd-headless.etcd-system.svc.cluster.local:2379
+ unset -v ETCDCTL_ENDPOINTS
+ etcdctl --user 'root:puIV%G!QRU1%vjII' endpoint health --endpoints=etcd-0.etcd-headless.etcd-system.svc.cluster.local:2379
etcd-0.etcd-headless.etcd-system.svc.cluster.local:2379 is healthy: successfully committed proposal: took = 2.769557ms
+ echo 'Snapshotting the keyspace...'
Snapshotting the keyspace...
++ date -u +%Y-%m-%d_%H-%M
+ current_time=2021-03-11_12-10
+ unset -v ETCDCTL_ENDPOINTS
+ etcdctl --user 'root:puIV%G!QRU1%vjII' snapshot save /snapshots/db-2021-03-11_12-10 --endpoints=etcd-0.etcd-headless.etcd-system.svc.cluster.local:2379
Error: could not open /snapshots/db-2021-03-11_12-10.part (open /snapshots/db-2021-03-11_12-10.part: permission denied)
```

### PR DESCRIPTION
**Pull request by danfromtitan**
**Description of the change**

This change fixes #5759 by adding the same _volume-permissions_ init container from the statefulset into the snapshot cronjob, while slightly adapting the conditions when that should happen. 

**Benefits**

Capture snapshots when security context enabled. The outcome is snapshots directory ownership is changed to allow runAsUser UID to write into it:

```bash
etcd 16:30:12.83 INFO  ==> Snapshotting the keyspace
[...]
Snapshot saved at /snapshots/db-2021-09-15_16-30
```

**Possible drawbacks**

None. 

**Applicable issues**

  - fixes  #5759

**Additional information**

.Values.volumePermissions.enabled is overloaded and should be removed from the chart, because since pod or container security context is enabled and PVC persistence is also enabled, then changing the directory permissions is implied. 

I left _volumePermissions_ unchanged because I wasn't sure whether there was another use case for it.

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/etcd/templates/cronjob.yaml [hunk 1]
```diff
@@ -30,6 +30,30 @@ spec:
           {{- if .Values.podSecurityContext.enabled }}
           securityContext: {{- omit .Values.podSecurityContext "enabled" | toYaml | nindent 12 }}
           {{- end }}
+          {{- if or .Values.initContainers .Values.volumePermissions.enabled }}
+          initContainers:
+            {{- if .Values.initContainers }}
+            {{- include "common.tplvalues.render" (dict "value" .Values.initContainers "context" $) | nindent 8 }}
+            {{- end }}
+            {{- if and .Values.volumePermissions.enabled (or .Values.podSecurityContext.enabled .Values.containerSecurityContext.enabled) }}
```
Comment (juan131) [inline bitnami/etcd/templates/cronjob.yaml:38]: The `initContainers` parameter is intended for **etcd** pods exclusively:
```suggestion
          {{- if and .Values.volumePermissions.enabled (or .Values.podSecurityContext.enabled .Values.containerSecurityContext.enabled) }}
          initContainers:
```
Feel free to add a new parameter  (e.g. `disasterRecovery.initcontainers`) if you find it useful but do not reuse the existing one.

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks so much for the contribution! Please check my comments when you have a chance.

File: bitnami/etcd/templates/cronjob.yaml (hunk 1 above)
Comment (danfromtitan) [reply bitnami/etcd/templates/cronjob.yaml]: You're right, .Values.initContainers doesn't make sense in the context of the snapshotter cronjob, I removed it.

**Review by juan131 — APPROVED** [review]
LGTM

**Pull request merged**